src/mlpack/tests/decision_tree_regressor_test.cpp

```diff
@@ -0,0 +1,1047 @@
+//  * categorical dataset using weights, with low-weight noise added.
+//  */
+// TEST_CASE("CategoricalInformationGainWeightedBuildTest", "[DecisionTreeTest]")
+// {
+//   arma::mat d;
+//   arma::Row<size_t> l;
+//   data::DatasetInfo di;
+//   MockCategoricalData(d, l, di);
+
+//   // Split into a training set and a test set.
+//   arma::mat trainingData = d.cols(0, 1999);
+//   arma::mat testData = d.cols(2000, 3999);
+//   arma::Row<size_t> trainingLabels = l.subvec(0, 1999);
+//   arma::Row<size_t> testLabels = l.subvec(2000, 3999);
+
+//   // Now create random points.
+//   arma::mat randomNoise(4, 2000);
+//   arma::Row<size_t> randomLabels(2000);
+//   for (size_t i = 0; i < 2000; ++i)
+//   {
+//     randomNoise(0, i) = math::Random();
+//     randomNoise(1, i) = math::Random();
+//     randomNoise(2, i) = math::RandInt(4);
+//     randomNoise(3, i) = math::RandInt(2);
+//     randomLabels[i] = math::RandInt(5);
+//   }
+
+//   // Generate weights.
+//   arma::rowvec weights(4000);
+//   for (size_t i = 0; i < 2000; ++i)
+//     weights[i] = math::Random(0.9, 1.0);
+//   for (size_t i = 2000; i < 4000; ++i)
+//     weights[i] = math::Random(0.0, 0.001);
+
+//   arma::mat fullData = arma::join_rows(trainingData, randomNoise);
+//   arma::Row<size_t> fullLabels = arma::join_rows(trainingLabels, randomLabels);
+
+//   // Build the tree.
+//   DecisionTree<InformationGain> tree(fullData, di, fullLabels, 5, weights, 10);
+
+//   // Now evaluate the accuracy of the tree.
+//   arma::Row<size_t> predictions;
+//   tree.Classify(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+//   size_t correct = 0;
+//   for (size_t i = 0; i < testData.n_cols; ++i)
+//     if (testLabels[i] == predictions[i])
+//       ++correct;
+
+//   // Make sure we got at least 70% accuracy.
+//   const double correctPct = double(correct) / double(testData.n_cols);
+//   REQUIRE(correctPct > 0.70);
+// }
+
+/**
+ * Test that the decision tree generalizes reasonably.
+ */
+TEST_CASE("SimpleGeneralizationTest_", "[DecisionTreeRegressorTest]")
+{
+
+  // Loading data.
+  data::DatasetInfo info;
+  arma::mat trainData, testData;
+  arma::Row<double> trainLabels, testLabels;
+  arma::rowvec weights = arma::ones<arma::rowvec>(355);
+  LoadBostonHousingDataset(trainData, testData, trainLabels, testLabels, info);
+
+  // Build decision tree.
+  DecisionTreeRegressor<MSEGain> d(trainData, info, trainLabels);
+
+  // Get the predicted test labels.
+  arma::Row<double> predictions;
+  d.Predict(testData, predictions);
+
+  REQUIRE(predictions.n_elem == testData.n_cols);
+
+  // Figure out rmse.
+  double rmse = RMSE(predictions, testLabels);
+
+  // REQUIRE(rmse < 9.21);
+  // std::cout << predictions << std::endl << testLabels;
+  arma::Row<double> trainPred;
+  d.Predict(trainData, trainPred);
+  // std::cout << trainPred;
+
+  std::cout << "Train RMSE: " << RMSE(trainLabels, trainPred) << std::endl;
```

Comment:
I imagine you are planning to modify this test, but if you want to keep it, we should definitely have a `REQUIRE()` test for the RMSE value. :+1:

Reply:
Yeah, this is one of the tests that is still left. Once it is completed, it will have a REQUIRE() statement.